### PR TITLE
[DEV-1168] wallet & user box

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: src/api/definitions/kyc.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export enum KycStatus {
+  NA = 'NA',
+  CHATBOT = 'Chatbot',
+  ONLINE_ID = 'OnlineId',
+  VIDEO_ID = 'VideoId',
+  CHECK = 'Check',
+  MANUAL = 'Manual',
```
Comment (davidleomay) [inline src/api/definitions/kyc.ts:7]: KYC status `Manual` doesn't exist anymore.

File: src/api/definitions/user.ts [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
-  kycDataComplete?: boolean;
-  ref?: string;
+  mail: string;
+  kycDataComplete: boolean;
+  kycState: KycState;
+  kycStatus: KycStatus;
+  tradingLimit: UserTradingLimit;
+
+  ref: string;
+  usedRef: string;
```
Comment (davidleomay) [inline src/api/definitions/user.ts:13]: `usedRef` is not returned anymore.

File: src/api/definitions/user.ts [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
-  mail?: string;
-  kycDataComplete?: boolean;
-  ref?: string;
+  mail: string;
+  kycDataComplete: boolean;
+  kycState: KycState;
+  kycStatus: KycStatus;
+  tradingLimit: UserTradingLimit;
+
+  ref: string;
```
Comment (davidleomay) [inline src/api/definitions/user.ts:12]: `ref` might be undefined

File: src/components/edit/mail.edit.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  } = useForm<FormData>();
+  const { changeMail } = useUserContext();
+
+  async function saveUser({ email }: FormData): Promise<void> {
+    changeMail(email).then(onSubmit);
+  }
+
+  const rules = Utils.createRules({
+    email: Validations.Required && Validations.Mail,
+  });
```
Comment (davidleomay) [inline src/components/edit/mail.edit.tsx:31]: If you copied the `createRules` from payment page, rule "concat" should be use like
```ts
  const rules: any = createRules({
    mail: [Validations.Required, Validations.Mail],
  });
```

File: src/hooks/clipboard.hook.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+interface ClipboardInterface {
+  copy: (text: string) => Promise<void>;
+}
+
+export function useClipboard(): ClipboardInterface {
+  async function copy(text: string): Promise<void> {
+    console.log('TODO copy', text);
```
Comment (davidleomay) [inline src/hooks/clipboard.hook.ts:7]: Show a small notification or temporarily switch the button icon after copy click (maybe for v2, together with spinners 😉).

Comment (Krysh90) [reply src/hooks/clipboard.hook.ts:7]: yes totally

File: src/components/edit/mail.edit.tsx (hunk 4 above)
Comment (Krysh90) [reply src/components/edit/mail.edit.tsx:31]: that is why it was not working as expected ;) thanks

File: src/api/definitions/user.ts (hunk 2 above)
Comment (davidleomay) [reply src/api/definitions/user.ts:13]: He doesn't need to know IMO

**Review by davidleomay — APPROVED** [review]
Wait for response of Oli concerning used ref.

**Pull request merged**